### PR TITLE
Fixup internal symbols to be namespaced with ossl_ .

### PR DESCRIPTION
**Pull request by slontis**
A series of Partial fixes for #12964

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
Adding 'ossl_' to provider_get_capabilities cost me the best part of a day to figure out that there was already a symbol with this name.

File: crypto/asn1/asn1_lib.c [hunk 1]
```diff
@@ -411,8 +411,9 @@ unsigned char *ASN1_STRING_data(ASN1_STRING *x)
 }
 #endif
 
-char *sk_ASN1_UTF8STRING2text(STACK_OF(ASN1_UTF8STRING) *text, const char *sep,
-                              size_t max_len /* excluding NUL terminator */)
+char *ossl_sk_ASN1_UTF8STRING2text(STACK_OF(ASN1_UTF8STRING) *text,
```
Comment (levitte) [inline crypto/asn1/asn1_lib.c:414]: I would drop `sk_`...

File: crypto/ec/curve448/ed448.h [hunk 2]
```diff
@@ -198,10 +203,11 @@ c448_error_t curve448_point_decode_like_eddsa_and_mul_by_ratio(
  * x (out): The ECDH private key as in RFC7748
  * ed (in): The EdDSA private key
  */
-c448_error_t c448_ed448_convert_private_key_to_x448(
-                            OSSL_LIB_CTX *ctx,
-                            uint8_t x[X448_PRIVATE_BYTES],
-                            const uint8_t ed[EDDSA_448_PRIVATE_BYTES],
-                            const char *propq);
+c448_error_t
+ossl_c448_ed448_convert_private_key_to_x448(
```
Comment (levitte) [inline crypto/ec/curve448/ed448.h:207]: This is quite a mouthful!  (and yeah, I know there are worse examples...)

Comment (t8m) [reply crypto/ec/curve448/ed448.h:207]: I suppose the c448 prefix could be just dropped here and in other similarly named functions.

File: crypto/asn1/asn1_lib.c (hunk 1 above)
Comment (t8m) [reply crypto/asn1/asn1_lib.c:414]: > I would drop `sk_`...

That seems wrong to me as that really works with stack and not a single ASN1_UTF8STRING.

File: crypto/ec/curve448/ed448.h (hunk 2 above)
Comment (levitte) [reply crypto/ec/curve448/ed448.h:207]: It could be abbreviated to `ossl_ed448_privkey2x448` or `ossl_ed448_privkey_to_x448`

File: crypto/asn1/asn1_lib.c (hunk 1 above)
Comment (slontis) [reply crypto/asn1/asn1_lib.c:414]: I will leave it as it is..

File: crypto/ec/curve448/ed448.h (hunk 2 above)
Comment (slontis) [reply crypto/ec/curve448/ed448.h:207]: I prefer not to do this, as it ends up with name clashes..
 There are already ossl_ed448_sign & ossl_ed448_verify for example.

**Comment by slontis** [comment]
rebased..
@paulidale can you review this monstrosity?
Review of commits might make it easier...

File: crypto/ec/curve448/ed448.h [hunk 3]
```diff
@@ -114,15 +115,16 @@ c448_error_t c448_ed448_sign_prehash(
...
-                               const uint8_t *message, size_t message_len,
-                               uint8_t prehashed, const uint8_t *context,
-                               uint8_t context_len,
-                               const char *propq);
+c448_error_t
+ossl_c448_ed448_verify(OSSL_LIB_CTX *ctx,
+                       const uint8_t
+                       signature[EDDSA_448_SIGNATURE_BYTES],
+                       const uint8_t
+                       pubkey[EDDSA_448_PUBLIC_BYTES],
```
Comment (paulidale) [inline crypto/ec/curve448/ed448.h:123]: Can't say I like this indentation but there doesn't seem to be much alternative.

File: crypto/http/http_local.h [hunk 4]
```diff
@@ -13,18 +13,20 @@
...
-                                    const char *content_type, BIO *req_mem,
-                                    int maxline, unsigned long max_resp_len,
-                                    int timeout,
-                                    const char *expected_content_type,
-                                    int expect_asn1);
-int http_use_proxy(const char *no_proxy, const char *server);
-const char *http_adapt_proxy(const char *proxy, const char *no_proxy,
-                             const char *server, int use_ssl);
+BIO *ossl_http_asn1_item2bio(const ASN1_ITEM *it, const ASN1_VALUE *val);
+OSSL_HTTP_REQ_CTX
```
Comment (paulidale) [inline crypto/http/http_local.h:17]: blank line before perhaps?

File: doc/man7/provider-base.pod [hunk 5]
```diff
@@ -90,8 +90,8 @@ provider-base
  void provider_unquery_operation(void *provctx, int operation_id,
                                  const OSSL_ALGORITHM *algs);
  const OSSL_ITEM *provider_get_reason_strings(void *provctx);
- int provider_get_capabilities(void *provctx, const char *capability,
-                               OSSL_CALLBACK *cb, void *arg);
+ int ossl_prov_get_capabilities(void *provctx, const char *capability,
```
Comment (paulidale) [inline doc/man7/provider-base.pod:93]: ???? This is a function pointer not a real function.  Why the change??

I suspect it should be reverted in this file.

File: test/curve448_internal_test.c [hunk 6]
```diff
@@ -601,40 +601,42 @@ static int test_ed448(void)
...
-                                       NULL, 0, NULL))
+            || !TEST_true(ossl_ed448ph_sign(NULL, outsig, dohash(hashctx, phmsg1,
+                                            sizeof(phmsg1)), phpubkey1,
+                                            phprivkey1, NULL, 0, NULL))
             || !TEST_int_eq(memcmp(phsig1, outsig, sizeof(phsig1)), 0)
-            || !TEST_true(ED448ph_sign(NULL, outsig, dohash(hashctx, phmsg2,
-                                       sizeof(phmsg2)), phpubkey2, phprivkey2,
-                                       phcontext2, sizeof(phcontext2), NULL))
+            || !TEST_true(ossl_ed448ph_sign(NULL, outsig, dohash(hashctx, phmsg2,
+                                            sizeof(phmsg2)), phpubkey2,
```
Comment (paulidale) [inline test/curve448_internal_test.c:637]: Indentation looks off here, it should indent to the '(' of the `dohash` call.

**Review by paulidale — CHANGES_REQUESTED** [review]
A couple of suggestions, one indentation problem and something that needs reverting.
Otherwise this looks good.

File: doc/man7/provider-base.pod (hunk 5 above)
Comment (slontis) [reply doc/man7/provider-base.pod:93]: capabilities.c contains the real function - guess I got confused.. Renamed

**Review by paulidale — APPROVED** [review]

**Comment by slontis** [comment]
Rebased with no changes

**Comment by slontis** [comment]
Thanks for reviewing. Merged to master.
